Bottom: 012a6ebc68d401fb1deb19be2ca59a106affadcd
Top:    47b2a93e911b6e017fa8e0683c7ffbd2dcf8510c
Author: Ian Kent <raven@themaw.net>
Date:   2013-07-30 15:40:20 +0800

Refresh of autofs-5.0.7-add-dumpmap-to-file.patch

---

diff --git a/daemon/automount.c b/daemon/automount.c
index 52a7956..e724ca3 100644
--- a/daemon/automount.c
+++ b/daemon/automount.c
@@ -1933,7 +1933,7 @@ int main(int argc, char *argv[])
 	int logpri = -1;
 	unsigned ghost, logging, daemon_check;
 	unsigned dumpmaps, foreground, have_global_options;
-	const char *dumpnames = NULL;
+	const char *dump = NULL;
 	time_t timeout;
 	time_t age = time(NULL);
 	struct rlimit rlim;
@@ -2031,8 +2031,8 @@ int main(int argc, char *argv[])
 
 		case 'm':
 			dumpmaps = 1;
-			printf("optarg %s\n", optarg);
-			dumpnames = getstropt(optarg, opt);
+			if (optstring)
+				dump = getstropt(optarg, opt);
 			break;
 
 		case 'O':
@@ -2176,7 +2176,7 @@ int main(int argc, char *argv[])
 		master_list->nc = nc;
 
 		lookup_nss_read_master(master_list, 0);
-		master_show_mounts(master_list, dumpnames);
+		master_show_mounts(master_list, dump);
 		exit(0);
 	}
